src/control/BBTrajectories/BBTrajectory1D.tpp

```diff
@@ -0,0 +1,210 @@
+//
+// Created by rolf on 15-01-20.
+//
+
+#include "control/BBTrajectories/BBTrajectory1D.h"
+#include <cmath>
+#include <iostream>
+#include <chrono>
+template<class num>
+num BBTrajectory1D<num>::fullBrakePos(num pos, num vel, num accMax) noexcept {
+    num acc = vel <= 0 ? accMax : - accMax;
+    num t = - vel/acc; // (inverted) time needed to break to zero velocity
+    return pos + 0.5*vel*t; // position after breaking
+}
+template<class num>
+num BBTrajectory1D<num>::accelerateBrakePos(num pos0, num vel0, num vel1, num accMax) noexcept {
+    num acc1;
+    num acc2;
+    if (vel1 >= vel0) {
+        acc1 = accMax;
+        acc2 = - accMax;
+    }
+    else {
+        acc1 = - accMax;
+        acc2 = accMax;
+    }
+    num t1 = (vel1 - vel0)/acc1; // time to reach vel1
+    num pos1 = pos0 + (0.5*(vel0 + vel1)*t1); //position at which we reach vel1
+    num t2 = - vel1/acc2; // time to max break from vel1 to 0
+    return pos1 + (0.5*vel1*t2); // position we stop at after initiating maximal break at pos1
+
+}
+template<class num>
+void
+BBTrajectory1D<num>::triangularProfile(num initialPos, num initialVel, num finalPos, num maxAcc,
+        bool invertedSign) noexcept {
+    num acc = invertedSign ? maxAcc : - maxAcc;
+    //compute the final time difference at which we switch from accelerating to breaking
+    num sq = (acc*(finalPos - initialPos) + 0.5*initialVel*initialVel)/(acc*acc);
+    num brakeTime = sq > 0 ? sqrt(sq) : 0;
```

Comment:
The `s > 0` check should not be needed. sq should always be nonnegative

Reply:
Not true. if (finalPos - initialPos)*acc  term is larger than the other and has a negative sign it is definitely smaller than 0.